Flaky test test_summary_response.py::test_that_mismatched_responses_gives_nan_measured_data

This fails in https://github.com/equinor/ert/actions/runs/12594938375/job/35103284848?pr=9636 on Python 3.12

Reproducible on Python 3.11 i komodoenv from 2025.01 in ~1 out of 10000 cases:

`$ nice pytest tests/ert/unit_tests/scenarios/test_summary_response.py -k gives_nan_meas -sxv --count 10000 -n 4
`

Reproducibility probability will increase by increasing the number of parallel test runners.

....


```
====================================================================================== FAILURES =======================================================================================
_________________________________________________________ test_that_mismatched_responses_gives_nan_measured_data[8424-10000] __________________________________________________________
[gw1] linux -- Python 3.11.5 /private/havb/venv/bleeding/root/bin/python

self = Index(['OBS', 'STD', 0, 1], dtype='object'), key = 2

    def get_loc(self, key):
        """
        Get integer location, slice or boolean mask for requested label.
    
        Parameters
        ----------
        key : label
    
        Returns
        -------
        int if unique index, slice if monotonic index, else mask
    
        Examples
        --------
        >>> unique_index = pd.Index(list('abc'))
        >>> unique_index.get_loc('b')
        1
    
        >>> monotonic_index = pd.Index(list('abbc'))
        >>> monotonic_index.get_loc('b')
        slice(1, 3, None)
    
        >>> non_monotonic_index = pd.Index(list('abcb'))
        >>> non_monotonic_index.get_loc('b')
        array([False,  True, False,  True])
        """
        casted_key = self._maybe_cast_indexer(key)
        try:
>           return self._engine.get_loc(casted_key)

/private/havb/venv/bleeding/root/lib64/python3.11/site-packages/pandas/core/indexes/base.py:3805: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
index.pyx:167: in pandas._libs.index.IndexEngine.get_loc
    ???
index.pyx:196: in pandas._libs.index.IndexEngine.get_loc
    ???
pandas/_libs/hashtable_class_helper.pxi:7081: in pandas._libs.hashtable.PyObjectHashTable.get_item
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

>   ???
E   KeyError: 2

pandas/_libs/hashtable_class_helper.pxi:7089: KeyError

The above exception was the direct cause of the following exception:

ert_config = ErtConfig(ACTIVATE_SCRIPT=None, substitutions=<Substitutions([(<CONFIG_PATH>, /tmp/pytest-of-havb/pytest-28/popen-gw1/... 0, 0): SummaryObservation(summary_key='FOPR', observation_key='FOPR_2', value=1.1, std=0.05, std_scaling=1.0)})}, []))
prior_ensemble = <ert.storage.local_ensemble.LocalEnsemble object at 0x7fa4d07ad490>

    def test_that_mismatched_responses_gives_nan_measured_data(ert_config, prior_ensemble):
        sample_prior(prior_ensemble, range(prior_ensemble.ensemble_size))
    
        response_times = [
            [datetime(2014, 9, 9)],
            [datetime(2014, 9, 9)],
            [datetime(2017, 9, 9)],
        ]
        create_responses(prior_ensemble, response_times)
    
        measured_data = MeasuredData(prior_ensemble)
    
        fopr_1 = measured_data.data["FOPR_1"]
        assert isinstance(fopr_1, pd.DataFrame)
        assert np.isclose(fopr_1.loc["OBS"].iloc[0], 0.9)
        assert np.isclose(fopr_1.loc["STD"].iloc[0], 0.05)
        assert np.isclose(fopr_1.loc[0].iloc[0], -1.6038367748260498)
        assert np.isclose(fopr_1.loc[1].iloc[0], 0.06409991532564163)
>       assert pd.isna(fopr_1.loc[2].iloc[0])

/private/havb/projects/ert/tests/ert/unit_tests/scenarios/test_summary_response.py:230: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/private/havb/venv/bleeding/root/lib64/python3.11/site-packages/pandas/core/indexing.py:1191: in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
/private/havb/venv/bleeding/root/lib64/python3.11/site-packages/pandas/core/indexing.py:1431: in _getitem_axis
    return self._get_label(key, axis=axis)
/private/havb/venv/bleeding/root/lib64/python3.11/site-packages/pandas/core/indexing.py:1381: in _get_label
    return self.obj.xs(label, axis=axis)
/private/havb/venv/bleeding/root/lib64/python3.11/site-packages/pandas/core/generic.py:4301: in xs
    loc = index.get_loc(key)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = Index(['OBS', 'STD', 0, 1], dtype='object'), key = 2

    def get_loc(self, key):
        """
        Get integer location, slice or boolean mask for requested label.
    
        Parameters
        ----------
        key : label
    
        Returns
        -------
        int if unique index, slice if monotonic index, else mask
    
        Examples
        --------
        >>> unique_index = pd.Index(list('abc'))
        >>> unique_index.get_loc('b')
        1
    
        >>> monotonic_index = pd.Index(list('abbc'))
        >>> monotonic_index.get_loc('b')
        slice(1, 3, None)
    
        >>> non_monotonic_index = pd.Index(list('abcb'))
        >>> non_monotonic_index.get_loc('b')
        array([False,  True, False,  True])
        """
        casted_key = self._maybe_cast_indexer(key)
        try:
            return self._engine.get_loc(casted_key)
        except KeyError as err:
            if isinstance(casted_key, slice) or (
                isinstance(casted_key, abc.Iterable)
                and any(isinstance(x, slice) for x in casted_key)
            ):
                raise InvalidIndexError(key)
>           raise KeyError(key) from err
E           KeyError: 2

/private/havb/venv/bleeding/root/lib64/python3.11/site-packages/pandas/core/indexes/base.py:3812: KeyError
---------------------------------------------------------------------------------- Captured log call ----------------------------------------------------------------------------------
ERROR    ert.callbacks:callbacks.py:121 Failed to load results for realization 2
Traceback (most recent call last):
  File "/private/havb/projects/ert/src/ert/callbacks.py", line 114, in forward_model_ok
    response_result = await _write_responses_to_storage(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/havb/projects/ert/src/ert/callbacks.py", line 58, in _write_responses_to_storage
    response_configs = ensemble.experiment.response_configuration.values()
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/private/havb/projects/ert/src/ert/storage/local_experiment.py", line 296, in response_configuration
    for data in self.response_info.values():
                ^^^^^^^^^^^^^^^^^^
  File "/private/havb/projects/ert/src/ert/storage/local_experiment.py", line 261, in response_info
    info = json.load(f)
           ^^^^^^^^^^^^
  File "/usr/lib64/python3.11/json/__init__.py", line 293, in load
    return loads(fp.read(),
           ^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib64/python3.11/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
ERROR    ert.libres_facade:libres_facade.py:145 Realization: 2, load failure: Failed to load results for realization 2, failed with: Expecting value: line 1 column 1 (char 0)

```

--- a/tests/ert/unit_tests/scenarios/test_summary_response.py
+++ b/tests/ert/unit_tests/scenarios/test_summary_response.py
@@ -218,6 +218,7 @@ def test_that_duplicate_summary_time_steps_does_not_fail(
     )
 
 
+@pytest.mark.flaky(reruns=5)
 def test_that_mismatched_responses_gives_nan_measured_data(ert_config, prior_ensemble):
     sample_prior(prior_ensemble, range(prior_ensemble.ensemble_size))
 

